test/extended/pods/oom_killer.go

```diff
@@ -0,0 +1,206 @@
+			g.Fail(fmt.Sprint("Expected atleast one OOM event but did not get one"))
+		}
+		hasStressPodBeenOOMKilled := false
+		// All the events messages should point to the same victim process, ideally the one which induced OOM
+		for _, oomEventMsg := range oomEventMsgList {
+			if strings.Contains(oomEventMsg, "stress") {
+				hasStressPodBeenOOMKilled = true
+			}
+		}
+		if !hasStressPodBeenOOMKilled {
+			result.Flakef("From events: expected oom inducing pod to be oomkilled but other pod oom killed")
+		}
+
+		testDuration := fmt.Sprintf("%v", exutil.DurationSinceStartInSeconds().Seconds()) + "s"
+		out, err := oc.Run("adm", "node-logs").Args(pod.Spec.NodeName, "--path=journal",
+			"--case-sensitive=false", "--grep=invoked oom-killer", "--since=-"+testDuration).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		hasStressPodBeenOOMKilledFromNodeLogs := false
+		if strings.Contains(strings.ToLower(out), "stress") {
+			hasStressPodBeenOOMKilledFromNodeLogs = true
+		}
+		if !hasStressPodBeenOOMKilledFromNodeLogs {
+			result.Flakef("From Nodelogs: expected oom inducing pod to be oomkilled but other pod oom killed")
+		}
+	})
+})
+
+var _ = g.Describe("[sig-node][Serial]", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewDefaultFramework("pod-eviction")
+	oc := exutil.NewCLIWithFramework(f)
+
+	// The goal of this test to verify if the OOM inducing pod is the one that is getting OOM Killed
+	g.It("should evict the same pod that causes memory crunch", func() {
+		currentTime := exutil.SuiteStartTime()
+		g.By("creating the pod that stresses memory")
+		podClient := f.PodClient()
+		ns := f.Namespace.Name
+		podName := naming.GetPodName("mem-stress-pod", string(uuid.NewUUID()))
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:  containerName,
+					Image: "alexeiled/stress-ng",
+					Args:  []string{"--vm", "2", "--vm-bytes", " 24G", "--timeout", "100m"},
+				},
+				},
+			},
+		}
+		pod = podClient.Create(pod)
+		err := e2epod.WaitForPodNameRunningInNamespace(oc.KubeFramework().ClientSet, pod.Name, oc.Namespace())
+		framework.ExpectNoError(err)
+		defer func() {
+			g.By("deleting the pod")
+			podClient.Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(0))
+		}()
+
+		pod, err = podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// To wait for pod to start hogging memory
+		time.Sleep(2 * time.Minute)
+
+		// Collect the events to see if we're getting OOM Killed.
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			g.Fail(fmt.Sprintf("Unexpected error: %v", err))
+		}
+
+		errorMsg := "The node was low on resource: memory"
+		evictionMsgList := []string{}
+		framework.Logf(" events %v", len(events.Items))
+		for _, event := range events.Items {
```

Comment:
you may want to extract this into a test helper

Reply:
The below if conditions have changed now. Perhaps it's not needed.